### DISCUSSION TITLE
GrpcApiError - "Failed to execute grpc AEDT command: GetProperties"

### DISCUSSION THREAD

**Discussion by chamanth-vct** [ansys/pyaedt#5318] [Q&A] [answered]
Hi,

I frequently encounter the following error while using pyaedt:

```txt
GrpcApiError: Failed to execute grpc AEDT command: GetProperties
```

Unfortunately, I have been unable to determine the exact cause of this issue. Additionally, I have not been successful in reproducing the problem with a sample file.

### Request:

Could you provide any guidance or insights into:

- Potential causes for this error.

- Best practices for debugging such issues.

- Any known scenarios where this might occur.

Any help would be greatly appreciated!

**Comment by Samuelopez-ansys** [comment]
Hi @chamanth-vct ,

I suggest using an IDE like PyCharm or VS to debug your code, then it will be easy to find where this error is happening. For instance, in PyCharm, I can see all places where GetProperties is used:

![image](https://github.com/user-attachments/assets/b8bc517f-a917-4413-b89a-88db302cd009)

It is difficult to predict where this is happening for you, so please try to reproduce it and send a snipped code.

**Comment by chamanth-vct** [comment]
Hi @Samuelopez-ansys 

Thank you for your suggestion. I was able to locate the source of the error. It appears when attempting to access the material name of an object attribute using the following code:

```py
desktop = ansys.aedt.core.Desktop()
prj = desktop.load_project(file_path)
model = prj.modeler
object_names = model.object_names

for item in object_names:
    geo_obj = model.get_object_from_name(item)
    try:
        geo_obj.material_name
    except Exception as e:
        print(item)
        print(f"\t{e.__class__.__name__}: {str(e)}")
```
**Error Message:**
```txt
RadiatingSurface_1
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_2
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_3
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_4
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_5
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_6
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties
RadiatingSurface_7
        GrpcApiError: Failed to execute grpc AEDT command: GetProperties

```

Unfortunately, I'm unable to share the specific model where this issue arises. However, I did notice an interesting detail: the error only occurs for certain objects that are included in the `object_names` list but are not visible in the GUI.

**Comment by chamanth-vct** [comment, marked as answer]
This issue is resolved at https://github.com/ansys/pyaedt/issues/5452
